sycl/test/basic_tests/esimd/flat_load.cpp

```diff
@@ -0,0 +1,43 @@
+// RUN: %clangxx -fsycl -fsycl-explicit-simd -fsycl-device-only -fsyntax-only -Xclang -verify %s
+// expected-no-diagnostics
+
+#include <CL/sycl.hpp>
+#include <limits>
+#include <sycl_esimd.hpp>
+#include <utility>
+
+using namespace sycl::intel::gpu;
+using namespace cl::sycl;
+
+void kernel(accessor<int, 1, access::mode::read_write, access::target::global_buffer> &buf) __attribute__((sycl_device)) {
+  simd<uint32_t, 32> offsets(0, 1);
+  simd<int, 32> v1(0, 1);
+
+  auto v0 = gather<int, 32>(buf.get_pointer(), offsets);
+
+  v0 = v0 + v1;
+
+  scatter<int, 32>(buf.get_pointer(), v0, offsets);
+}
+
+void kernel(accessor<uint8_t, 1, access::mode::read_write, access::target::global_buffer> &buf) __attribute__((sycl_device)) {
+  simd<uint32_t, 32> offsets(0, 1);
+  simd<uint8_t, 32> v1(0, 1);
+
+  simd<uint8_t, 32> v0 = gather<uint8_t, 32>(buf.get_pointer(), offsets);
+
+  //v0 = v0 + v1;
```

Comment:
Why is this commented out?

Reply:
This code is commented out because right now `simd<uint8_t, 32> + simd<uint8_t, 32>` yields `simd<int, 32>` and `operator=` fails. This happens because overloaded `operator+` computes common type for LHR and RHS which happens to be `simd<int, 32>` according to promotion rules (`uint8_t + uint8_t` yeilds `int`).

Reply:
As agreed with @cmc-rep, resolved in the following way:
```
v0 = convert<uint8_t>(v0 + v1);
```